com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -266,14 +270,72 @@ internal string GetSceneNameFromPath(string scenePath)
         }
 
         /// <summary>
-        /// Generates the scenes in build list
+        /// Generates the hash values and associated tables
+        /// for the scenes in build list
         /// </summary>
         internal void GenerateScenesInBuild()
         {
-            ScenesInBuild.Clear();
+            HashToBuildIndex.Clear();
+            BuildIndexToHash.Clear();
             for (int i = 0; i < SceneManager.sceneCountInBuildSettings; i++)
             {
-                ScenesInBuild.Add(GetSceneNameFromPath(SceneUtility.GetScenePathByBuildIndex(i)));
+                var scenePath = SceneUtility.GetScenePathByBuildIndex(i);
+                var hash = XXHash.Hash32(scenePath);
+                var buildIndex = SceneUtility.GetBuildIndexByScenePath(scenePath);
+                HashToBuildIndex.Add(hash, buildIndex);
+                BuildIndexToHash.Add(buildIndex, hash);
+            }
+        }
+
+        /// <summary>
+        /// Gets the scene name from a hash value generated from the full scene path
+        /// </summary>
+        internal string SceneNameFromHash(uint sceneHash)
+        {
+            // In the event there is no scene associated with the scene event
+            // (this can happen) then just return "No Scene";
+            if (sceneHash == 0)
```

Comment:
what about `InvalidSceneNameOrPath` then? I'm confused. Is `-1` the invalid value or `0`? also hashes are `unsigned`, so better use `uint` for `InvalidSceneNameOrPath` too?

Reply:
I updated the comments for this check for better clarity:
            // This can happen during unit tests when clients first connect and the only scene loaded is the 
            // unit test scene (which is ignored by default) that results in a scene event that has no associated
            // scene.  Under this specific special case, we just return "No Scene".